datasets/de/abgeordnetenwatch/crawler.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import json
+import math
+from zavod import Context
+from zavod import helpers as h
+
+
+def crawl(context: Context):
+    # Fetch the source data URL specified in the metadata to a local path:
+    api_response = context.fetch_json(context.dataset.data.url)
+    total_results = api_response["meta"]["result"]["total"]
+    num_batches = math.ceil(total_results / 100)
+
+    bi = 0
+    while bi < num_batches:
+        context.log.info(
+            "Get Politicians from range {} to {}".format(bi * 100, (bi + 1) * 100)
+        )
+        if bi > 0:
+            api_response = context.fetch_json(
+                context.dataset.data.url, params={"range_start": bi * 100}
+            )
+            batch_total_results = api_response["meta"]["result"]["total"]
+            if batch_total_results != total_results:
+                num_batches = math.ceil(batch_total_results / 100)
+
+        for candidate in api_response.get("data"):
+            parliament_info = candidate["parliament_period"]["label"].split(" ")
+            state = parliament_info[0]
+            inception_date = parliament_info[1]
+            dissolution_date = parliament_info[3]
+
+            position = h.make_position(
+                context,
+                name="Member of Parliament ({} - {}) - Germany/{}".format(
+                    inception_date, dissolution_date, state
+                ),
+                country="Germany",
+                subnational_area=state,
+                inception_date=inception_date,
+                dissolution_date=dissolution_date,
+            )
+
+            person = context.make("Person")
+            person.id = context.make_id(candidate["politician"]["id"])
+            h.apply_name(person, full=candidate["politician"]["label"])
+            h.make_occupancy(context, person, position, False)
+            context.emit(person)
+            context.emit(position)
```

Comment:
Aha - the idea is that make_occupancy() will only return an occupancy entity if the position is within the allowable time period. Currently if it ended more than 5 years ago it won't return an Occupancy entity.

So you should check if an occupancy was returned, and only emit the person, position and occupancy if an occupancy was created.

Reply:
Thanks! Fixed on my last push and now the targets file contains both politicians and their positions